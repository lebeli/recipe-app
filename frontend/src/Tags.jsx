import React from "react";
import TextField from "@material-ui/core/TextField";
import Autocomplete from "@material-ui/lab/Autocomplete";
import "./Tags.scss";

function Tags(params) {
  return (
    <div className="Tags">
      <Autocomplete
        multiple
        options={params.ingredients}
        getOptionLabel={ingredient => ingredient}
        filterSelectedOptions
        onChange={(_, tags) => {
          params.updateState("chosen_ingredients", tags);
        }}
        renderInput={tags => (
          <TextField
            {...tags}
            className="tag"
            variant="outlined"
            placeholder="Nach Zutaten suchen..."
            margin="normal"
            fullWidth
          />
        )}
      />
    </div>
  );
}

export default Tags;
