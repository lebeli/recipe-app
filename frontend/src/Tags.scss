@import "./styles/styles.scss";

.Tags {
  max-width: 600px;
  .MuiInputBase-root {
    background-color: white;
    border-radius: 24px;
    padding: 0 0 0 5px;
  }

  .MuiAutocomplete-inputRoot[class*="MuiOutlinedInput-root"] {
    padding: 0 10px 0 10px;
  }

  .MuiOutlinedInput-notchedOutline {
    border: 0px;
    border-radius: 24px;
    height: 50px;
  }

  .MuiChip-root {
    background-color: $primaryColor;
    color: white;
    position: relative;

    .MuiSvgIcon-root {
      fill: white;
    }

    .MuiChip-label {
      font-family: $secondaryFont;
    }
  }

  .MuiInputBase-input {
    font-family: $secondaryFont;
  }

  .MuiIconButton-root {
    background-color: transparent;
  }
}
