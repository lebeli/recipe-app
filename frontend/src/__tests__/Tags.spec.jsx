import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import Tags from "../Tags";

describe("<Tags />", () => {
  test("should render elements", () => {
    const wrapper = mount(
      <Tags
        ingredients={[
          "Gurke",
          "Tomate",
          "Zwiebel",
          "Knoblauch",
          "Ziegenk채se",
          "Gouda",
          "Erdn체sse"
        ]}
        chosen_ingredients={[]}
      />
    );
    expect(wrapper.find(".Tags"));
    expect(wrapper.find(".tag").length).toEqual(5);
  });

  test("renders correctly", () => {
    const tree = renderer
      .create(
        <Tags
          ingredients={[
            "Gurke",
            "Tomate",
            "Zwiebel",
            "Knoblauch",
            "Ziegenk채se",
            "Gouda",
            "Erdn체sse"
          ]}
          chosen_ingredients={[]}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
