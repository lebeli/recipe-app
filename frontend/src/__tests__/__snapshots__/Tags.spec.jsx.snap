// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags /> renders correctly 1`] = `
<div
  className="Tags"
>
  <div
    aria-expanded={false}
    aria-owns={null}
    className="MuiAutocomplete-root"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
  >
    <div
      className="MuiFormControl-root MuiTextField-root tag MuiFormControl-marginNormal MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        onClick={[Function]}
      >
        <input
          aria-activedescendant={null}
          aria-autocomplete="list"
          aria-controls={null}
          aria-invalid={false}
          autoCapitalize="none"
          autoComplete="off"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          placeholder="Nach Zutaten suchen..."
          required={false}
          spellCheck="false"
          type="text"
          value=""
        />
        <div
          className="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Clear"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
          </button>
          <button
            aria-label="Open"
            className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Open"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </span>
          </button>
        </div>
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-83 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-84"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</div>
`;
